src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/FrameworkHandlerAdapter.cs

```diff
@@ -30,14 +29,14 @@ internal sealed class FrameworkHandlerAdapter : IFrameworkHandle
     private readonly IMessageBus _messageBus;
     private readonly VSTestBridgedTestFrameworkBase _adapterExtensionBase;
     private readonly TestSessionContext _session;
-    private readonly IClientInfo _clientInfo;
     private readonly CancellationToken _cancellationToken;
     private readonly bool _isTrxEnabled;
     private readonly MessageLoggerAdapter _comboMessageLogger;
     private readonly string _testAssemblyPath;
+    private readonly IServiceProvider _serviceProvider;
 
-    public FrameworkHandlerAdapter(VSTestBridgedTestFrameworkBase adapterExtensionBase, TestSessionContext session, IClientInfo clientInfo, string[] testAssemblyPaths,
-        ITestApplicationModuleInfo testApplicationModuleInfo, ILoggerFactory loggerFactory, IMessageBus messageBus, IOutputDevice outputDevice,
+    public FrameworkHandlerAdapter(VSTestBridgedTestFrameworkBase adapterExtensionBase, TestSessionContext session, string[] testAssemblyPaths,
```

Comment:
My idea was to avoid (when possible) to pas the `IServiceProvider` because it's hiding the real deps (I don't know what to mock and what are the deps of the object looking at the ctor).